projects/plugins/jetpack/modules/comments/subscription-modal-on-comment/subscription-modal.js

```diff
@@ -0,0 +1,173 @@
+document.addEventListener( 'DOMContentLoaded', function () {
+	const modal = document.getElementsByClassName( 'jetpack-subscription-modal' )[ 0 ];
+
+	if ( ! modal ) {
+		return;
+	}
+
+	const close = document.getElementsByClassName( 'jetpack-subscription-modal__close' )[ 0 ];
+	const subscribeForms = document.querySelectorAll( '.jetpack-subscription-modal__form' );
+
+	let redirectUrl = '';
+	let subscriptionData = '';
+	let hasLoaded = false;
+
+	function reloadOnCloseSubscriptionModal( customUrl ) {
+		const destinationUrl = customUrl ? new URL( customUrl ) : new URL( redirectUrl );
+
+		// Prevent redirect to external sites.
+		if ( destinationUrl.hostname !== window.location.hostname ) {
+			return;
+		}
+
+		localStorage.setItem( 'jetpack-subscription-modal-goto', destinationUrl.hash );
```

Comment:
I'd add more specificity to name not to confuse with the other "subscribe modal" functionality.